src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -1,48 +1,235 @@
+      
+        //armPID.setTolerance(posToleranceRad, velToleranceRadPSec);
+
+        armPIDMaster.setFeedbackDevice(armMasterEncoder);
+        armPIDMaster.setPositionPIDWrappingEnabled(true);
+        armPIDMaster.setPositionPIDWrappingMinInput(LOWER_ANGLE_LIMIT);
+        armPIDMaster.setPositionPIDWrappingMaxInput(UPPER_ANGLE_LIMIT);
+        //two PIDs?
+        
+        SmartDashboard.putData("Arm", this);
+
+        setPoint = getCurrentArmState();
+        goalState = getCurrentArmState();
+        setArmTarget(goalState.position);
     }
-    
+
     @Override
     public void periodic() {
-     // run moveArm passing in controller here
-		}
-}
+
+        if(DriverStation.isDisabled()) resetGoal();
+
+        //ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD = SmartDashboard.getNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        // armConstraints = new TrapezoidProfile.Constraints(MAX_FF_VEL , MAX_FF_ACCEL );
+        
+
+        //smart dahsboard stuff
+        //SmartDashboard.putBoolean("ArmPIDAtSetpoint", armPID1.atSetpoint());
+       // SmartDashboard.putBoolean("ArmProfileFinished", armProfile.isFinished(armProfileTimer.get()));
+        //posToleranceRad = SmartDashboard.getNumber("Arm Tolerance Pos", posToleranceRad);
+        //velToleranceRadPSec= SmartDashboard.getNumber("Arm Tolerance Vel", velToleranceRadPSec);
+
+       // SmartDashboard.putNumber("MaxHoldingTorque", maxHoldingTorqueNM());
+        //SmartDashboard.putNumber("V_PER_NM", getV_PER_NM());
+       // SmartDashboard.putNumber("COMDistance", getCoM().getNorm());
+        SmartDashboard.putNumber("InternalArmVelocity", armMasterEncoder.getVelocity());
+        //SmartDashboard.putNumber("Arm Current", armMotor.getOutputCurrent());
+
+       // SmartDashboard.putNumber("ArmPos", getArmPos());
+
+        autoCancelArmCommand();
+        driveArm();
+    }
+
+  public void autoCancelArmCommand() {
+        if(!(getDefaultCommand() instanceof ArmTeleop) || DriverStation.isAutonomous()) return;
+
+        double requestedSpeeds = ((ArmTeleop) getDefaultCommand()).getRequestedSpeeds();
+
+        if(requestedSpeeds != 0) {
+            Command currentArmCommand = getCurrentCommand();
+            if(currentArmCommand != getDefaultCommand() && currentArmCommand != null) {
+                currentArmCommand.cancel();
+            }
+        }
+    }
+
+    //#region Drive Methods
+    private void driveArm(){
+      
+      setPoint = armProfile.calculate(kDt, setPoint, goalState);
+      double armFeedVolts = armFeed.calculate(goalState.position, goalState.velocity);
+      if ((getArmPos() < LOWER_ANGLE_LIMIT && getCurrentArmGoal().velocity > 0) || (getArmPos() > UPPER_ANGLE_LIMIT && getCurrentArmGoal().velocity > 0)){
+        armFeedVolts = armFeed.calculate(getCurrentArmGoal().position, 0);
+      }
+      armPIDMaster.setReference(setPoint.position, CANSparkBase.ControlType.kVelocity, 0, armFeedVolts);
```

Comment:
when the pid is within tolerance, you may want to disable pid to prevent arm from shaking a lot. Also, if arm goes beyond limits you probably don't want PID to run, you only want feedforward to maintain its position and not move.